### PR TITLE
[Enhancement](tvf) Add authentication for workload group tvf

### LINKED ISSUE #21322: [Enhancement](tvf) Add authentication for workload group tvf
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, displaying workload group information by tvf is not authenticated, and all workload group information is displayed. 
We create workload group g1 by root, and perform select operation as a non-root user.
MySQL [(none)]> select * from workload_groups();
+-------+--------+--------------------------+------------+
| Id          | Name    | Item                                      | Value           |
+-------+--------+--------------------------+------------+
| 10002 | normal   | enable_memory_overcommit | true          |
| 10002 | normal   | max_concurrency               | 2147483647|
| 10002 | normal   | queue_timeout                    | 0                   |
| 10002 | normal   | max_queue_size                 | 0                   |
| 10002 | normal   | memory_limit                       | 30%             |
| 10002 | normal   | cpu_share                            | 10                 |
| 10082 | g1          | enable_memory_overcommit | true           |
| 10082 | g1         | max_concurrency                 | 2147483647|
| 10082 | g1         | queue_timeout                      | 0                   |
| 10082 | g1         | max_queue_size                   | 0                   |
| 10082 | g1         | memory_limit                         | 30%             |
| 10082 | g1         | cpu_share                              | 10                 |
+-------+--------+--------------------------+------------+

We expect to show only information of workload groups to which the user has permission.
MySQL [(none)]> select * from workload_groups();
+-------+--------+--------------------------+------------+
| Id          | Name    | Item                                      | Value           |
+-------+--------+--------------------------+------------+
| 10002 | normal   | enable_memory_overcommit | true          |
| 10002 | normal   | max_concurrency               | 2147483647|
| 10002 | normal   | queue_timeout                    | 0                   |
| 10002 | normal   | max_queue_size                 | 0                   |
| 10002 | normal   | memory_limit                       | 30%             |
| 10002 | normal   | cpu_share                            | 10                 |
+-------+--------+--------------------------+------------+

### Solution

Add authentication for workload group tvf.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yongjinhou**
## Proposed changes

Issue Number: close #21322

Add authentication for workload group tvf.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yongjinhou** [comment]
run buildall

**Review by luozenglin — DISMISSED** [review]
LGTM

File: be/src/vec/exec/scan/vmeta_scan_node.cpp [hunk 1]
```diff
@@ -65,6 +65,7 @@ Status VMetaScanNode::_init_scanners(std::list<VScannerSPtr>* scanners) {
     for (auto& scan_range : _scan_ranges) {
         std::shared_ptr<VMetaScanner> scanner = VMetaScanner::create_shared(
                 _state, this, _tuple_id, scan_range, _limit_per_scanner, runtime_profile());
+        scanner->set_user_identity(_scan_params.current_user_ident);
```
Comment (luozenglin) [inline be/src/vec/exec/scan/vmeta_scan_node.cpp:68]: Need to check that _scan_params.current_user_ident exists

File: be/src/vec/exec/scan/vmeta_scanner.cpp [hunk 2]
```diff
@@ -78,6 +77,10 @@ Status VMetaScanner::prepare(RuntimeState* state, const VExprContextSPtrs& conju
     return Status::OK();
 }
 
+void VMetaScanner::set_user_identity(TUserIdentity user_identity) {
```
Comment (luozenglin) [inline be/src/vec/exec/scan/vmeta_scanner.cpp:80]:
```suggestion
void VMetaScanner::set_user_identity(const TUserIdentity& user_identity) {
```

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.54 seconds
 stream load tsv:          461 seconds loaded 74807831229 Bytes, about 154 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          70.6 seconds inserted 10000000 Rows, about 141K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230629083203_clickbench_pr_169733.html

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 42.62 seconds
 stream load tsv:          457 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          69.7 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230629095538_clickbench_pr_169846.html

**Comment by yongjinhou** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.24 seconds
 stream load tsv:          457 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          68.3 seconds inserted 10000000 Rows, about 146K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230629112949_clickbench_pr_169862.html

**Comment by yongjinhou** [comment]
run buildall

**Review by luozenglin — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 42.69 seconds
 stream load tsv:          460 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          68.8 seconds inserted 10000000 Rows, about 145K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230629125549_clickbench_pr_169960.html

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p1

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**